cuda_core/cuda/core/experimental/_event.pyx

```diff
@@ -165,6 +165,65 @@ cdef class Event:
                 raise CUDAError(err)
             raise RuntimeError(explanation)
 
+    def __hash__(self) -> int:
+        """Return hash based on the underlying CUevent handle address and context.
+
+        This enables Event objects to be used as dictionary keys and in sets.
+        Two Event objects wrapping the same underlying CUDA event will hash
+        to the same value and be considered equal.
+
+        Returns
+        -------
+        int
+            Hash value based on the event handle address and context handle.
+
+        Notes
+        -----
+        Includes the context handle in the hash to prevent collisions when
+        handles are reused across different contexts. While handles are
+        context-scoped and typically not reused across contexts, including
+        the context provides defense-in-depth against hash collisions.
+
+        Warning
+        -------
+        Using a closed or destroyed event as a dictionary key or in a set
+        results in undefined behavior. The event handle may be reused by
+        the CUDA driver for new events.
+        """
+        # Context should always be set as a post-condition of Event construction
+        return hash((type(self), <uintptr_t>(int(self._ctx_handle)), <uintptr_t>(self._handle)))
```

Comment:
Since `self._ctx_handle` is a `Context` object which is now hashable, I think we can just pass it directly instead of needing to convert it to `int` and cast to `<uintptr_t>`?